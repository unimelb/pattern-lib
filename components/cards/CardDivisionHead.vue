<template>
  <a
    :href="href"
    :class="{ 'card--division--head-compact': compact }"
    class="btn-owner card card--division card--division--head card--bdr bg-white">
    <div class="card__thumb">
      <div
        :style="`background-image: url('${thumb}')`"
        class="card__thumb-img" />
    </div>
    <div class="card__inner">
      <div class="card__subheader">
        <h3 class="card__header">{{ name }}</h3>
        <p class="card__meta">{{ title }}</p>
      </div>
      <hr v-if="!compact">
      <slot />
      <ButtonIcon
        v-if="cols === 1"
        class="btn--wide">Contacts and bio</ButtonIcon>
    </div>
    <div
      v-if="cols !== 1 && compact === false"
      class="card__footer">
      <ButtonIcon
        element="span"
        class="btn--fullwidth">Contacts and bio</ButtonIcon>
    </div>
  </a>
</template>

<script>
export default {
  props: {
    thumb: {
      type: String,
      default: '',
    },
    cols: {
      type: Number,
      default: 3,
    },
    title: {
      type: String,
      default: '',
    },
    name: {
      type: String,
      default: '',
      required: true,
    },
    href: {
      type: String,
      default: '',
      required: true,
    },
    compact: {
      type: Boolean,
      default: false,
    },
  },
};
</script>
