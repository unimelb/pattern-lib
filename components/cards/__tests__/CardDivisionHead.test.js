import {
  shallow,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import CardDivisionHead from '../CardDivisionHead.vue';

expect.extend(toHaveNoViolations);

describe('CardDivisionHead', () => {
  it('should match snapshot', () => {
    const result = shallow(CardDivisionHead, {
      propsData: {
        name: 'mock name',
        href: 'www.unimelb.edu.au',
      },
    }).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = shallow(CardDivisionHead, {
      propsData: {
        name: 'mock name',
        href: 'www.unimelb.edu.au',
      },
    }).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
