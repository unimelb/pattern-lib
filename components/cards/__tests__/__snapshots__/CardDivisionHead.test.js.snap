// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDivisionHead should match snapshot 1`] = `
<a
  class="btn-owner card card--division card--division--head card--bdr bg-white"
  href="www.unimelb.edu.au"
>
  <div
    class="card__thumb"
  >
    <div
      class="card__thumb-img"
      style="background-image: url();"
    />
  </div>
   
  <div
    class="card__inner"
  >
    <div
      class="card__subheader"
    >
      <h3
        class="card__header"
      >
        mock name
      </h3>
       
      <p
        class="card__meta"
      >
        
      </p>
    </div>
     
    <hr />
      
    <!---->
  </div>
   
  <div
    class="card__footer"
  >
    <buttonicon
      class="btn--fullwidth"
      element="span"
    >
      Contacts and bio
    </buttonicon>
  </div>
</a>
`;
